sources/packages/backend/libs/services/src/msfaa-number/msfaa-number-shared.service.ts

```diff
@@ -136,19 +185,45 @@ export class MSFAANumberSharedService {
           updatedAt: now,
         },
       );
-      // Create the new MSFAA record for the student.
-      const newMSFAANumber = new MSFAANumber();
-      newMSFAANumber.msfaaNumber = await this.consumeNextSequence(
-        offeringIntensity,
-      );
-      newMSFAANumber.student = { id: studentId } as Student;
-      newMSFAANumber.referenceApplication = {
-        id: referenceApplicationId,
-      } as Application;
-      newMSFAANumber.offeringIntensity = offeringIntensity;
-      newMSFAANumber.creator = { id: auditUserId } as User;
-      newMSFAANumber.createdAt = now;
-      await entityManager.getRepository(MSFAANumber).save(newMSFAANumber);
+      // The reactivated msfaaNumber.
+      let newActiveMSFAA = {
```

Comment:
can't we do something like this `let newActiveMSFAA = options?.existingMSFAA` ?

Reply:
@ann-aot I can do something like `let newActiveMSFAA = options?.existingMSFAA as MSFAANumber;` because without casting it as `MSFAANumber`, in the `if` block assignments like `newActiveMSFAA.student = { id: studentId } as Student;` fail as it is otherwise typed as `let newActiveMSFAA: Pick<MSFAANumber, "id" | "msfaaNumber" | "dateSigned" | "serviceProviderReceivedDate">`